test_scripts/API/VehicleData/HandsOffSteering/common.lua

```diff
@@ -0,0 +1,369 @@
+--! @parameters:
+--! @pRpcName: RPC name
+--! @pResultCode: resultCode value for expectation on App
+--! pAppId: application number (1, 2, etc.)
+--! @return: none
+--]]
+function m.appAlreadySubUnsubscribed(pRpcName, pResultCode, pAppId)
+  if not pAppId then pAppId = 1 end
+  local cid = m.getMobileSession(pAppId):SendRPC(pRpcName, { handsOffSteering = true })
+  m.getHMIConnection():ExpectRequest("VehicleInfo." .. pRpcName, { handsOffSteering = true }) :Times(0)
+  m.getMobileSession(pAppId):ExpectResponse(cid, { success = false, resultCode = "IGNORED",
+    handsOffSteering = {dataType = "VEHICLEDATA_HANDSOFFSTEERING", resultCode = pResultCode }, })
+  m.getMobileSession(pAppId):ExpectNotification("OnHashChange") :Times(0)
+end
+
+--[[ @ignitionOff: IGNITION_OFF sequence
+--! @parameters: none
+--! @return: none
+--]]
+function m.ignitionOff()
+  config.ExitOnCrash = false
+  local timeout = 5000
+  local function removeSessions()
+    for i = 1, m.getAppsCount() do
+      m.deleteSession(i)
+    end
+  end
+  local event = events.Event()
+  event.matches = function(event1, event2) return event1 == event2 end
+  EXPECT_EVENT(event, "SDL shutdown")
+  :Do(function()
+      removeSessions()
+      StopSDL()
+      config.ExitOnCrash = true
+    end)
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "SUSPEND" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLPersistenceComplete")
+  :Do(function()
+      m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications",{ reason = "IGNITION_OFF" })
+      for i = 1, m.getAppsCount() do
+        m.getMobileSession(i):ExpectNotification("OnAppInterfaceUnregistered", { reason = "IGNITION_OFF" })
+      end
+    end)
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  :Times(m.getAppsCount())
+  local isSDLShutDownSuccessfully = false
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+      utils.cprint(35, "SDL was shutdown successfully")
+      isSDLShutDownSuccessfully = true
+      RAISE_EVENT(event, event)
+    end)
+  :Timeout(timeout)
+  local function forceStopSDL()
+    if isSDLShutDownSuccessfully == false then
+      utils.cprint(35, "SDL was shutdown forcibly")
+      RAISE_EVENT(event, event)
+    end
+  end
+  RUN_AFTER(forceStopSDL, timeout + 500)
+end
+
+--[[ @registerAppSuccessWithResumption: Successful application registration with custom expectations for resumption
+--! @parameters:
+--! pAppId - application number (1, 2, etc.)
+--! isHMISubscription: if false SDL resumes SubscribeVehicleData for App2 and does not send
+--! VehicleInfo.SubscribeVehicleData request to HMI
+--! @return: none
+--]]
+function m.registerAppSuccessWithResumption(pAppId, isHMISubscription)
+  if not pAppId then pAppId = 1 end
+  local handsOffSteeringResponseData = {
```

Comment:
Please add `handsOffSteeringResponseData` in local varialbe in common file and use it